packages/@glimmer/runtime/lib/vm/rehydrate-builder.ts

```diff
@@ -379,15 +379,21 @@ export class RehydrateBuilder extends NewElementBuilder implements ElementBuilde
   __pushRemoteElement(
     element: SimpleElement,
     cursorId: string,
-    nextSibling: Option<SimpleNode> = null
+    insertBefore: Option<null>
   ): Option<RemoteLiveBlock> {
     let marker = this.getMarker(element as HTMLElement, cursorId);
 
     if (marker.parentNode === element) {
+      if (insertBefore !== null) {
+        while (element.lastChild !== marker) {
+          element.removeChild(element.lastChild!);
+        }
+      }
+
       let currentCursor = this.currentCursor;
       let candidate = currentCursor!.candidate;
 
-      this.pushElement(element, nextSibling);
+      this.pushElement(element, null);
```

Comment:
I think we should pass through `insertBefore` even though we know _currently_ its always `null`...

```suggestion
      this.pushElement(element, insertBefore);
```